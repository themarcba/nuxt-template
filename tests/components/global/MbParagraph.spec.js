import {} from 'jest';
// Doc: https://vue-test-utils.vuejs.org/api/
import { createLocalVue, config, shallowMount } from '@vue/test-utils';
import MbParagraph from '@/components/global/MbParagraph.vue';
import NuxtStubs from '@/tests/nuxt-stubs';

const localVue = createLocalVue();

// Doc: https://vue-test-utils.vuejs.org/api/#config
config.showDeprecationWarnings = true;

describe('MbParagraph.vue', () => {
  const wrapper = shallowMount(MbParagraph, {
    localVue,
    stubs: NuxtStubs
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('matches the snapshot', async () => {
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ secondary: true });
    expect(wrapper).toMatchSnapshot();
  });
});
