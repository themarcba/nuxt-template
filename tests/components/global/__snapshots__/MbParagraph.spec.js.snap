// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MbParagraph.vue matches the snapshot 1`] = `<p class="my-4 text-primary"></p>`;

exports[`MbParagraph.vue matches the snapshot 2`] = `<p class="my-4 text-secondary"></p>`;
